### PR TITLE
Centos8 and latest

### PR DESCRIPTION
**Pull request by dominikholler**
@rmohr @ksimon1 Can you please have a look?
Please find an example image in quay.io/dholler/centos .

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -21,14 +21,16 @@ function build() {
     source ${distro}/interface.sh
     SHASUM=$(containerdisks::needs_update ${tag})
     if [ -n "${SHASUM}" ]; then
-      docker build -t quay.io/containerdisks/${distro}:${tag} -t quay.io/containerdisks/${distro}:${tag}-${TAG_SUFFIX} --label "shasum=${SHASUM}" -f ${distro}/${tag}/Dockerfile .
+      docker build -t quay.io/containerdisks/${distro}:${tag} -t quay.io/containerdisks/${distro}:${tag}-${TAG_SUFFIX} -t latest --label "shasum=${SHASUM}" -f ${distro}/${tag}/Dockerfile .
```
Comment (rmohr) [inline build.sh:24]: @dominikholler I would prefer if we don't introduce `latest`. Are you sure that it is needed?

File: centos/8/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM centos as builder
```
Comment (rmohr) [inline centos/8/Dockerfile:1]: Could you stick with `alpine`? It is sooo much smaller, making development and build experience much better.

File: build.sh (hunk 1 above)
Comment (rmohr) [reply build.sh:24]: Like I would definitely want to do a conscious move from centos 8 to centos 9 and `8` and `9` are already moving tags.

File: centos/8/Dockerfile (hunk 2 above)
Comment (dominikholler) [reply centos/8/Dockerfile:1]: I agree.
I had some trouble consuming https://cloud.centos.org/centos/8/x86_64/images/CHECKSUM in alpine. Do you have an idea how to check it in alpine?

File: build.sh (hunk 1 above)
Comment (dominikholler) [reply build.sh:24]: I am fine, so we have to document that the tag is used explicitly, e.g. quay.io/dholler/centos:8 .

File: centos/8/Dockerfile (hunk 2 above)
Comment (rmohr) [reply centos/8/Dockerfile:1]: `apk add wget` probably solves it. The `wget` included in alpine by default is not really `wget`.

Comment (dominikholler) [reply centos/8/Dockerfile:1]:
```
[dominik@p1gen3 containerdisks]$ podman run --rm -t -i alpine
/ # apk add wget
(1/3) Installing libunistring (0.9.10-r1)
(2/3) Installing libidn2 (2.3.2-r0)
(3/3) Installing wget (1.21.2-r2)
Executing busybox-1.34.1-r3.trigger
OK: 8 MiB in 17 packages
/ # cd /tmp/
/tmp # wget https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream
-GenericCloud-8-20200113.0.x86_64.qcow2
--2021-11-25 13:43:55--  https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2
Resolving cloud.centos.org (cloud.centos.org)... 3.8.21.190, 2a05:d01c:c6a:cc01:11ae:4a43:febc:bd33
Connecting to cloud.centos.org (cloud.centos.org)|3.8.21.190|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 739962368 (706M) [application/octet-stream]
Saving to: 'CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2'

CentOS-Stream-Gener 100%[===================>] 705.68M  29.2MB/s    in 26s     

2021-11-25 13:44:21 (27.4 MB/s) - 'CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2' saved [739962368/739962368]

/tmp # wget https://cloud.centos.org/centos/8-stream/x86_64/images/CHECKSUM
--2021-11-25 13:44:30--  https://cloud.centos.org/centos/8-stream/x86_64/images/CHECKSUM
Resolving cloud.centos.org (cloud.centos.org)... 3.8.21.190, 2a05:d01c:c6a:cc01:11ae:4a43:febc:bd33
Connecting to cloud.centos.org (cloud.centos.org)|3.8.21.190|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4008 (3.9K)
Saving to: 'CHECKSUM'

CHECKSUM            100%[===================>]   3.91K  --.-KB/s    in 0s      

2021-11-25 13:44:30 (61.4 MB/s) - 'CHECKSUM' saved [4008/4008]

/tmp # grep CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2 CHECKSUM
# CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2: 722624K 
SHA256 (CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2) = b80ca9ccad8715aa1aeee582906096d3190b91cebe05fcb3ac7cb197fd68a7fe
/tmp # grep CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2 CHECKSUM > c2
/tmp # sha256sum -c c2
sha256sum: WARNING: 2 of 2 computed checksums did NOT match

```

Comment (rmohr) [reply centos/8/Dockerfile:1]: I think that you are feeding this line into the sha256sum:

```
SHA256 (CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2) = b80ca9ccad8715aa1aeee582906096d3190b91cebe05fcb3ac7cb197fd68a7fe
```

Is this something which `s256sum` can understand? As far as I know this is the format it needs:

```
b80ca9ccad8715aa1aeee582906096d3190b91cebe05fcb3ac7cb197fd68a7fe  CentOS-Stream-GenericCloud-8-20200113.0.x86_64.qcow2
```

Note that there are two spaces between the shasum and the filename.

Comment (dominikholler) [reply centos/8/Dockerfile]: I just learned that there is a BSD style checksum and a GNU style one. Looks like CentOS is using the BSD style one, while alpine/busybox requires the GNU style one. So let's sed adjust the format.

File: centos/interface.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+function containerdisks::needs_update() {
+  local name=centos
+  local tag="$1"
+  local base_url="https://cloud.centos.org/centos/${tag}/x86_64/images/"
+
+  local os_image="$(wget -O - "${base_url}" | grep "GenericCloud.*qcow2" | sed -e 's/.*>\(.*qcow2\)<.*/\1/' | sort -r | head -n 1)"
+  local repo="quay.io/containerdisks/centos"
+
+
+  local latest_shasum="$(wget -O - "${base_url}CHECKSUM" | grep ${os_image} | tail -n 1 | grep -o '[a-Z0-9]*$')"
```
Comment (rmohr) [inline centos/interface.sh:14]: I see that something is not right with this check as the tests report an empty shasum:

```
Latest shasum for centos:8: 
```

Comment (dominikholler) [reply centos/interface.sh:14]: For me this works. Do you have a hint how I can reproduce the issue?

**Comment by rmohr** [comment]
/close

Thanks for reviewing #8  which makes this PR obsolet.

**Pull request closed without merging**